Cubical/Foundations/GroupoidLaws.agda

```diff
@@ -278,3 +312,190 @@ hcomp-cong : ∀ {ℓ} {A : Set ℓ} {φ} → (u : I → Partial φ A) → (u0 :
              (ueq : ∀ i → PartialP φ (\ o → u i o ≡ u' i o)) → (outS u0 ≡ outS u0') [ φ ↦ (\ { (φ = i1) → ueq i0 1=1}) ]
              → (hcomp u (outS u0) ≡ hcomp u' (outS u0')) [ φ ↦ (\ { (φ = i1) → ueq i1 1=1 }) ]
 hcomp-cong u u0 u' u0' ueq 0eq = inS (\ j → hcomp (\ i o → ueq i o j) (outS 0eq j))
+
+
+
+cube-comp₋₀₋ :
+    (c : I → I → I → A)
+  → {a' : Square _ _ _ _}
+  → (λ i i₁ → c i i0 i₁) ≡ a'
+  → (I → I → I → A)
+cube-comp₋₀₋ c p i j k =
+   hcomp
+     (λ l → λ {
+        (i = i0) → c i0 j k
+       ;(i = i1) → c i1 j k
+       ;(j = i0) → p l i k
+       ;(j = i1) → c i i1 k
+       ;(k = i0) → c i j i0
+       ;(k = i1) → c i j i1
+      })
+    (c i j k)
+
+cube-comp₀₋₋ :
+    (c : I → I → I → A)
+  → {a' : Square _ _ _ _}
+  → (λ i i₁ → c i0 i i₁) ≡ a'
+  → (I → I → I → A)
+cube-comp₀₋₋ c p i j k =
+   hcomp
+     (λ l → λ {
+        (i = i0) → p l j k
+       ;(i = i1) → c i1 j k
+       ;(j = i0) → c i i0 k
+       ;(j = i1) → c i i1 k
+       ;(k = i0) → c i j i0
+       ;(k = i1) → c i j i1
+      })
+    (c i j k)
+
+
+
+pentagonIdentity : (p : x ≡ y) → (q : y ≡ z) → (r : z ≡ w) → (s : w ≡ v)
+                      →
+            (assoc p q (r ∙ s) ∙ assoc (p ∙ q) r s)
+                              ≡
+   cong (p ∙_) (assoc q r s) ∙∙ assoc p (q ∙ r) s ∙∙ cong (_∙ s) (assoc p q r)
+
+pentagonIdentity {x = x} {y} p q r s =
+        (λ i →
+              (λ j → cong (p ∙_) (assoc q r s) (i ∧ j))
+           ∙∙ (λ j → lemma₀₀ i j ∙ lemma₀₁ i j)
+           ∙∙ (λ j → lemma₁₀ i j ∙ lemma₁₁ i j)
+        )
+   where
+
+
+    lemma₀₀ : ( i j : I) → _ ≡ _
```

Comment:
Wouldn't this be easier to read if you gave the endpoints?

Reply:
Or are they too big?

Reply:
at last one end of each of those lemmas is big and non generic, should i write only one then, or both even if they are big and non informative ?

Reply:
Nah, let's leave it like this then